Kernel/Syscalls/unveil.cpp

```diff
@@ -90,6 +84,33 @@ int Process::sys$unveil(Userspace<const Syscall::SC_unveil_params*> user_params)
         }
     }
 
+    // Now, let's try and resolve the path and obtain custody of the inode on the disk, and if not, bail out with
+    // the error from resolve_path_without_veil()
+    // However, if the user specified unveil() with "c" permissions, we don't set errno if ENOENT is encountered,
+    // because they most likely intend the program to create the file for them later on.
+    // If this case is encountered, the parent node of the path is returned and the custody of that inode is used instead.
+    RefPtr<Custody> parent_custody; // Parent inode in case of ENOENT
+    String new_unveiled_path;
+    auto custody_or_error = VFS::the().resolve_path_without_veil(path.value(), root_directory(), &parent_custody);
+    if (!custody_or_error.is_error()) {
+        new_unveiled_path = custody_or_error.value()->absolute_path();
+    } else if (custody_or_error.error() == -ENOENT && (new_permissions & UnveiledPath::Access::CreateOrRemove)) {
+        String basename = LexicalPath(path.value()).basename();
+        new_unveiled_path = String::formatted("{}/{}", parent_custody->absolute_path(), basename);
+    } else {
+        // FIXME Should this be EINVAL?
+        return custody_or_error.error();
+    }
+
+    RefPtr<Custody> custody;
+    if (custody_or_error.is_error()) {
+        custody = parent_custody;
+        new_unveiled_path = parent_custody->absolute_path();
+    } else {
+        custody = custody_or_error.value();
+        new_unveiled_path = custody->absolute_path();
+    }
```

Comment:
Wait, you don't need to do any of this. The only thing you need is to work out `new_unveiled_path`, and that is already done above.

Reply:
Whoops, forgot to remove that.